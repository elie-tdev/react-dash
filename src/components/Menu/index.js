export { default as MobileMenu } from './MobileMenu';
export { default as MobileMenuItem } from './MobileMenuItem';
