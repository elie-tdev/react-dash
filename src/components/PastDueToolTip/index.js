export { default as PastDueToolTip } from './PastDueToolTip';
