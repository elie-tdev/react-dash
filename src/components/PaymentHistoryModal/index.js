export { default as PaymentHistoryModal } from './PaymentHistoryModal';
