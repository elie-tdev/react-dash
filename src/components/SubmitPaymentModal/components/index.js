export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
