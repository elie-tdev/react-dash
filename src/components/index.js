export * from './LoanGlance';
export * from './Button';
export * from './Header';
export * from './Footer';
export * from './LabelValueColumn';
export * from './LabelValueRow';
export * from './Loader';
export * from './CenteredScreenContent';
export * from './PaymentView';
export * from './Pagination';
export * from './Screen';
export * from './LoanSelect';
export * from './StatementGlance';
export * from './Menu';
export * from './PaymentHistoryModal';
export * from './PaymentStatus';
export * from './PaymentType';
export * from './SubmitPaymentModal';
export * from './GridContainer';
export * from './PastDueToolTip';
