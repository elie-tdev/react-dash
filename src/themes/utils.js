export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
