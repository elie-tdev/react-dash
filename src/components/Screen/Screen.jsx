import { Box, makeStyles } from '@material-ui/core';
import { useContext, useEffect, useMemo, useState } from 'react';
import TimeAgo from 'javascript-time-ago';
import cn from 'classnames';
import { useParams } from 'react-router';

import { Header, Footer } from '@/components';
import { useGlobalStyles } from '@/themes';
import { MenuContext } from '@/contexts';

// setTimeout() accepts up to a 32 bit number. Greater overflows and instantly executes
const SET_TIMEOUT_MAX_SAFE_INTERVAL = 2147483647;
function getSafeTimeoutInterval(interval) {
  return Math.min(interval, SET_TIMEOUT_MAX_SAFE_INTERVAL);
}

const useStyles = makeStyles(theme => ({
  elevatedBox: {
    paddingTop: theme.spacing(2),
    [theme.breakpoints.up('lg')]: {
      paddingTop: theme.spacing(5),
      background: theme.palette.common.gradient.primary.lightGrey,
    },
  },
  withMinHeight: {
    minHeight: '75vh',
  },
}));

const Screen = ({
  title,
  children,
  lastUpdated,
  elevated = true,
  showHeader = true,
  showPaddingTop = true,
  className,
}) => {
  const globalStyles = useGlobalStyles();
  const styles = useStyles();
  const {
    actions: { setRouteParams },
  } = useContext(MenuContext);

  const timeAgo = useMemo(() => new TimeAgo('en-US'), []);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const [formattedDate, setFormattedDate] = useState(null);

  useEffect(() => {
    if (!lastUpdated) return;

    const [newFormattedDate, timeToNextUpdate] = timeAgo.format(lastUpdated, {
      getTimeToNextUpdate: true,
    });
    setFormattedDate(newFormattedDate);
    const updateTimer = setTimeout(() => {
      const [newFormattedDate] = timeAgo.format(lastUpdated, {
        getTimeToNextUpdate: true,
      });
      setFormattedDate(newFormattedDate);
    }, getSafeTimeoutInterval(timeToNextUpdate || 60 * 1000));
    return () => clearTimeout(updateTimer);
  }, [formattedDate, lastUpdated]);

  const routeParams = useParams();

  useEffect(() => {
    setRouteParams(routeParams);
  }, [routeParams]);

  return (
    <Box>
      {showHeader && <Header title={title} />}
      <Box
        className={cn(
          elevated ? globalStyles.screenPaddingOn : globalStyles.screenPaddingOff,
          showPaddingTop && styles.elevatedBox,
          styles.withMinHeight,
          className,
        )}
      >
        {children}
      </Box>
      <Footer />
    </Box>
  );
};

export default Screen;
