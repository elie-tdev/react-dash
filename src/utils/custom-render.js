import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
