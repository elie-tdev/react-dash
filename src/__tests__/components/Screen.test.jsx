import React from 'react';

import { Screen } from '@/components/Screen';
import { render, waitFor } from '@/utils/custom-render';

describe('Screen Component', () => {
  test('renders without loan select', async () => {
    const { container } = render(<Screen title="Hello" loanSelect={true} />);
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
