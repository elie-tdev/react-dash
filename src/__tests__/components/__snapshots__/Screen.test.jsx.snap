// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Component renders without loan select 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-32"
  >
    <div
      class="MuiBox-root MuiBox-root-54 makeStyles-headerContainer-35 makeStyles-row-8"
      data-testid="Header"
    >
      <div
        class="MuiContainer-root makeStyles-container-33"
      >
        <div
          class="MuiBox-root MuiBox-root-55 makeStyles-row-8 makeStyles-headerContentBox-36"
        >
          <div
            class="MuiBox-root MuiBox-root-56 makeStyles-menuIconContainer-42"
            data-testid="MenuIconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-menuIcon-41"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root MuiBox-root-57 makeStyles-logoBox-50 makeStyles-logoBox-53"
          >
            <img
              class="makeStyles-logo-51"
              src="logo.png"
            />
          </div>
          <h2
            class="MuiTypography-root makeStyles-title-34 makeStyles-title-52 MuiTypography-h2"
          >
            Customer Dashboard
          </h2>
        </div>
        <div
          class="MuiBox-root MuiBox-root-64 makeStyles-column-10"
          data-testid="LoanSelectButton"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-65 makeStyles-headerSpacer-39"
    />
    <div
      class="MuiBox-root MuiBox-root-66 makeStyles-headerBottomWrapp-37 makeStyles-screenPaddingOn-21 makeStyles-row-8"
    >
      <div
        class="MuiContainer-root makeStyles-headerBottomContainer-38"
      >
        <div
          class="MuiBox-root MuiBox-root-67 makeStyles-row-8"
        >
          <div
            class="MuiBox-root MuiBox-root-77 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-78 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-bottomMenuIcon-46"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Loan Overview
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-79 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-80 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-bottomMenuIcon-46"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Balance Detail
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-81 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-82 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-bottomMenuIcon-46"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Payment History
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-83 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-84 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-bottomMenuIcon-46"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Statements
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-85 makeStyles-row-8 makeStyles-menuItemsRightBox-44"
        >
          <div
            class="MuiBox-root MuiBox-root-86 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-87 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-menuIcon-41"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Preferences
            </p>
          </div>
          <div
            class="MuiBox-root MuiBox-root-88 makeStyles-row-8 makeStyles-desktopMenuItem-68 makeStyles-leftMenu-69"
            data-testid="NavigationMenuItem-desktop"
          >
            <div
              class="MuiBox-root MuiBox-root-89 makeStyles-iconContainer-72 makeStyles-column-10 makeStyles-icon-75"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-menuIcon-41"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-itemName-73 MuiTypography-body1"
            >
              Log Out
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-90 makeStyles-menuIconRightBox-45"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-menuIcon-41"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-91 makeStyles-headerBottomSpacer-40"
    />
    <div
      class="MuiBox-root MuiBox-root-92 makeStyles-screenPaddingOn-21 makeStyles-elevatedBox-30 makeStyles-withMinHeight-31"
    />
    <div
      class="MuiBox-root MuiBox-root-112 makeStyles-root-93"
      data-testid="Footer"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainer-113 makeStyles-gridContainer-114 makeStyles-mainGridContainer-104 MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-desktopLogoContainer-103 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-3"
          >
            <img
              class="makeStyles-logo-94 makeStyles-logo-115"
              src="logo.png"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
          >
            <div
              class="MuiBox-root MuiBox-root-116 makeStyles-menuBox-107"
            >
              <h4
                class="MuiTypography-root makeStyles-menuHeadTitle-95 MuiTypography-h4"
              >
                FreedomPlus
              </h4>
              <div
                class="MuiBox-root MuiBox-root-117 makeStyles-menuList-109"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-101 MuiTypography-colorPrimary"
                  href="https://www.freedomplus.com/privacypolicy/"
                  rel="noopener"
                  target="_blank"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-119 MuiButton-contained makeStyles-menuItemButton-102 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
                    data-testid="SecondaryButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h4
                        class="MuiTypography-root makeStyles-menuItem-96 MuiTypography-h4"
                      >
                        Privacy Notice
                      </h4>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-101 MuiTypography-colorPrimary"
                  href="https://www.freedomplus.com/california-privacy-notice/"
                  rel="noopener"
                  target="_blank"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-120 MuiButton-contained makeStyles-menuItemButton-102 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
                    data-testid="SecondaryButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h4
                        class="MuiTypography-root makeStyles-menuItem-96 MuiTypography-h4"
                      >
                        California Privacy Notice
                      </h4>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-101 MuiTypography-colorPrimary"
                  href="https://www.freedomplus.com/termsofuse"
                  rel="noopener"
                  target="_blank"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-121 MuiButton-contained makeStyles-menuItemButton-102 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
                    data-testid="SecondaryButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h4
                        class="MuiTypography-root makeStyles-menuItem-96 MuiTypography-h4"
                      >
                        Terms of Use
                      </h4>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root makeStyles-divider-108"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
          >
            <div
              class="MuiBox-root MuiBox-root-122 makeStyles-menuBox-107"
            >
              <h4
                class="MuiTypography-root makeStyles-menuHeadTitle-95 MuiTypography-h4"
              >
                Consolidation Plus
              </h4>
              <div
                class="MuiBox-root MuiBox-root-123 makeStyles-menuList-109"
              >
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-101 MuiTypography-colorPrimary"
                  href="https://consolplus.com/privacy-notice/"
                  rel="noopener"
                  target="_blank"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-124 MuiButton-contained makeStyles-menuItemButton-102 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
                    data-testid="SecondaryButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h4
                        class="MuiTypography-root makeStyles-menuItem-96 MuiTypography-h4"
                      >
                        Privacy Notice
                      </h4>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
                <a
                  class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-101 MuiTypography-colorPrimary"
                  href="https://consolplus.com/california-privacy-notice/"
                  rel="noopener"
                  target="_blank"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-125 MuiButton-contained makeStyles-menuItemButton-102 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
                    data-testid="SecondaryButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h4
                        class="MuiTypography-root makeStyles-menuItem-96 MuiTypography-h4"
                      >
                        California Privacy Notice
                      </h4>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root makeStyles-divider-108"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
          >
            <div
              class="MuiBox-root MuiBox-root-126 makeStyles-menuBox-107"
            >
              <h4
                class="MuiTypography-root makeStyles-menuHeadTitle-95 MuiTypography-h4"
              >
                Mailing Address
              </h4>
              <div
                class="MuiBox-root MuiBox-root-127 makeStyles-menuList-109"
              >
                <h4
                  class="MuiTypography-root makeStyles-menuDescription-97 MuiTypography-h4"
                >
                  Main Office
                </h4>
                <h4
                  class="MuiTypography-root makeStyles-menuDescription-97 MuiTypography-h4"
                >
                  2114 E Rio Salado Blvd
                </h4>
                <h4
                  class="MuiTypography-root makeStyles-menuDescription-97 MuiTypography-h4"
                >
                  Tempe, AZ 85281
                </h4>
                <h4
                  class="MuiTypography-root makeStyles-menuDescription-97 MuiTypography-h4"
                >
                   PO Box 2340
                </h4>
                <h4
                  class="MuiTypography-root makeStyles-menuDescription-97 MuiTypography-h4"
                >
                  Phoenix, AZ 85002-2340
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainer-113 makeStyles-gridContainer-128 MuiGrid-container MuiGrid-spacing-xs-3"
        >
          <div
            class="MuiGrid-root MuiGrid-item makeStyles-menuBottomWrap-98 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
          >
            <div
              class="MuiBox-root MuiBox-root-129 makeStyles-bottomLogoBox-105"
            >
              <img
                class="makeStyles-logo-94 makeStyles-logo-130"
                src="logo.png"
              />
            </div>
            <div
              class="MuiBox-root MuiBox-root-131 makeStyles-bottomCopyrightBox-106"
            >
              <h4
                class="MuiTypography-root makeStyles-menuBottomDescription-99 MuiTypography-h4"
              >
                © 2021 Freedom Financial Asset Management, LLC. All Rights Reserved.
              </h4>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-132 MuiButton-contained makeStyles-button-100 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
              data-testid="SecondaryButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Contact Us
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
